### ISSUE TITLE
[RFC] Row virtualization overscanning tradeoff

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#270]
The current logic for over scanning is:

https://github.com/mui-org/material-ui-x/blob/b537d5d01fee06079eedbf3253a159907c9356d4/packages/grid/x-grid-modules/src/hooks/root/useContainerProps.ts#L50-L52

A couple of thoughts:

1. This seems counter-intuitive, from what I understand, over scanning is about avoiding white flashes. It seems that it should be correlated to the height of the rows and how quickly the end-user is scrolling, I don't see any relation with the height of the viewport.
2. We might want to provide an option to allow developer to customize the threshold. If we do, we might want to have a limit to the size of the buffer, rendering 10,000 rows can get ugly.

**Comment by dtassone** [comment]
viewportPageSize is calculated with Height of the grid and row height...

**Comment by oliviertassinari** [comment]
So if there are 20 rows displays, it overscan 20 rows. If there are 2 rows displayed, it overscan 2 rows. Is this accurate?

**Comment by dtassone** [comment]
Yes atm there is a factor 2 because of its design, with the rendering zone...
I wouldn't change this for now as I think we could improve the design of the rendering engine and also achieve more flexibility in the extra rendered rows

**Comment by oliviertassinari** [comment]
OK, this does sounds a lot better than having the overscan based on the height of the data grid viewport.
